### ISSUE TITLE
Compile error with 32-bit system

### ISSUE THREAD

**Issue by Michanord** [Theldus/wsServer#14]
Hi,
i was trying to use your library in a project at my university.
It runs fine and is easy to use and compile on my local system but my code has to run on a 32-bit ARM processor.

I get the following error during compilation:
```
.../wsserver-src/src/ws.c: In function ‘read_frame’:
.../wsserver-src/src/ws.c:813:30: error: left shift count >= width of type [-Werror=shift-count-overflow]
    (((size_t)next_byte(wfd)) << 56) | /* frame[2]. */
                              ^~
.../wsserver-src/src/ws.c:814:30: error: left shift count >= width of type [-Werror=shift-count-overflow]
    (((size_t)next_byte(wfd)) << 48) | /* frame[3]. */
                              ^~
.../wsserver-src/src/ws.c:815:30: error: left shift count >= width of type [-Werror=shift-count-overflow]
    (((size_t)next_byte(wfd)) << 40) | (((size_t)next_byte(wfd)) << 32) |
                              ^~
.../wsserver-src/src/ws.c:815:65: error: left shift count >= width of type [-Werror=shift-count-overflow]
    (((size_t)next_byte(wfd)) << 40) | (((size_t)next_byte(wfd)) << 32) |
                                                                 ^~
```
You seem to assume that `size_t` is 64-bits wide. That leads to issus for 32-bit systems.
Would it be possible to replace the use of `size_t` with `uint64_t` here? Or do you have any other idea how to fix this?

**Comment by Theldus** [comment]
Hi @Michanord,
Thanks for giving this project a try.

You're absolutely right. Although I actually tested wsServer against x86 and ARM processors, I forgot that both were 64-bit. Regarding the fix, yes, I think it's okay to switch to `uint64_t`. In fact, that seems the right way to go. Besides, I have to change `frame_length`, `frame_size`, and such.

I will fix this as well as test on these architectures. Thanks for spotting this.

**Comment by Theldus** [comment]
Hi @Michanord,
This issue was fixed in b2027a9, please let me know if this works for you.

**Comment by Michanord** [comment]
It now compiles without any problems and runs fine on the ARM device.
Thank you for your quick help!
